### PR TITLE
Use conda for travis-ci

### PR DESCRIPTION
**Pull request by chezou**
In order to speed up CI, I propose to use [Miniconda](http://conda.pydata.org/docs/travis.html).
Using Miniconda we can execute CI within [2 minitues](https://travis-ci.org/chezou/fastFM/builds/100968206).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,11 +1,11 @@
-language: c
+language: python
 
 python:
     - "2.7"
 
 os:
     - linux
-    - osx
+    #- osx
```
Comment (chezou) [inline .travis.yml]: Prevent to run OSX CI, I commented out this line.

File: .travis.yml [hunk 2]
```diff
@@ -1,11 +1,11 @@
-language: c
+language: python
```
Comment (chezou) [inline .travis.yml]: This change makes easy to test multiple version of python.

Comment (ibayer) [inline .travis.yml]: > Python builds are not available on the OSX environment.

https://docs.travis-ci.com/user/languages/python

File: .travis.yml (hunk 1 above)
Comment (ibayer) [inline .travis.yml]: Travis currently checks if the build compiles on OSX despite that we can't run the tests.
Please don't comment out this line.

**Comment by ibayer** [comment]
Using Miniconda is good idea we just need to make sure that Travis still runs the OSX build.
Thanks for the PR.

File: .travis.yml (hunk 2 above)
Comment (chezou) [inline .travis.yml]: Oops :(
But we can't test multiple version such as 2.7 and 3.5 without using `language: python`.

Comment (ibayer) [inline .travis.yml]: Yes, it's a pain...
Looks like this is a long standing issue (they discuss some workarounds):
https://github.com/travis-ci/travis-ci/issues/2312

**Comment by ibayer** [comment]
https://github.com/travis-ci/travis-ci/issues/2312

File: .travis.yml (hunk 2 above)
Comment (chezou) [inline .travis.yml]: OK, I'll try to use `language: c` with some dirty ways...

**Comment by chezou** [comment]
@ibayer I fixed using `language: c`. Could you review it?

File: .travis.yml [hunk 3]
```diff
@@ -11,19 +11,39 @@ dist: trusty
...
 install:
     - git submodule update --init --recursive
-    - pip install --user -r requirements.txt
-    - pip install --user nose
     - make
-    - pip install --user .
+    - pip install .
 
 script:
     # travis issue for osx, some libraries are build for x86_64 and others for i386.
```
Comment (ibayer) [inline .travis.yml]: Can we now run the tests for osx too?

Comment (chezou) [inline .travis.yml]: OK, I'll try it.

Comment (chezou) [inline .travis.yml]: waiting... https://travis-ci.org/ibayer/fastFM/builds/101049673

Comment (chezou) [inline .travis.yml]: CI passed! :tada:

**Comment by chezou** [comment]
CI passed both linux and Mac OS X with Miniconda! :tada: https://travis-ci.org/ibayer/fastFM/builds/101049673

**Comment by ibayer** [comment]
:thumbsup:
That the osx tests run now is awesome! I'm merging now. 
Thanks for the PR.

**Comment by chezou** [comment]
Thanks for your reviewing!

**Pull request merged**